src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTests.cs

```diff
@@ -2,272 +2,180 @@
-            a.Links = new[] { b, c, d }; // complete the cycle
-            yield return a;
-
-            // Structs
-            yield return new EmptyStruct();
-            yield return new StructWithIntField { X = 42 };
-            yield return new StructWithStringFields { String1 = "hello", String2 = "world" };
-            yield return new StructContainingOtherStructs { Nested1 = new StructWithStringFields { String1 = "a", String2 = "b" }, Nested2 = new StructWithStringFields { String1 = "3", String2 = "4" } };
-            yield return new StructContainingArraysOfOtherStructs();
-            yield return new StructContainingArraysOfOtherStructs { Nested = new StructContainingOtherStructs[0] };
-            var s = new StructContainingArraysOfOtherStructs
-            {
-                Nested = new[]
+                foreach (FormatterAssemblyStyle assemblyFormat in new[] { FormatterAssemblyStyle.Full, FormatterAssemblyStyle.Simple })
                 {
-                    new StructContainingOtherStructs { Nested1 = new StructWithStringFields { String1 = "a", String2 = "b" }, Nested2 = new StructWithStringFields { String1 = "3", String2 = "4" } },
-                    new StructContainingOtherStructs { Nested1 = new StructWithStringFields { String1 = "e", String2 = "f" }, Nested2 = new StructWithStringFields { String1 = "7", String2 = "8" } },
+                    foreach (TypeFilterLevel filterLevel in new[] { TypeFilterLevel.Full, TypeFilterLevel.Low })
+                    {
+                        foreach (FormatterTypeStyle typeFormat in new[] { FormatterTypeStyle.TypesAlways, FormatterTypeStyle.TypesWhenNeeded, FormatterTypeStyle.XsdString })
+                        {
+                            yield return new object[] { obj[0], assemblyFormat, filterLevel, typeFormat };
+                        }
+                    }
                 }
-            };
-            yield return s;
-            yield return new object[] { s, new StructContainingArraysOfOtherStructs?(s) };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidateBasicObjectsRoundtrip_MemberData))]
+        public void ValidateBasicObjectsRoundtrip(object obj, FormatterAssemblyStyle assemblyFormat, TypeFilterLevel filterLevel, FormatterTypeStyle typeFormat)
+        {
+            object clone = FormatterClone(obj, null, assemblyFormat, filterLevel, typeFormat);
+            if (!ReferenceEquals(obj, string.Empty)) // "" is interned and will roundtrip as the same object
+            {
+                Assert.NotSame(obj, clone);
+            }
 
-            // ISerializable
-            yield return new BasicISerializableObject(1, "2");
-            yield return new DerivedISerializableWithNonPublicDeserializationCtor(1, "2");
+            CheckForAnyEquals(obj, clone);
+        }
 
-            // Various other special cases
-            yield return new TypeWithoutNamespace();
+        //[Fact]
+        //public void Serialize()
```

Comment:
This is for generating serialized hashes, since we havent investigated all the issues yet we will leave it for now